lib/event_store_client/connection.rb

```diff
@@ -79,5 +73,47 @@ def client
           host: host, port: port, per_page: per_page
         )
     end
+
+    def read_from_stream(stream, direction:, start:, count:, resolve_links:)
+      response =
+        client.read(
+          stream, start: start, direction: direction, count: count, resolve_links: resolve_links
+        )
+      return [] if response.body.nil? || response.body.empty?
+      JSON.parse(response.body)['entries'].map do |entry|
+        deserialize_event(entry)
+      end.reverse
+    end
+
+    def read_all_from_stream(stream, resolve_links:)
+      count = 20
+      start = 0
+      events = []
+
+      loop do
```

Comment:
Do we handle any way if reading the whole stream break at some point? Let's say we've read some data and sth broke meanwhile(lost internet connection) what happened in this situation? - just question

Reply:
I'd avoid using infinite loops if possible. Please consider changing it to `while x do y`

Yes, we should handle failures, especially in this case - if there is a problem with reading data, we should be sure that request tries several times to repeat and only then fail